File: CLItask_Manager.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define TASK_FILE "tasks.txt"
#define MAX_TASK_LEN 100

typedef struct {
    char description[MAX_TASK_LEN];
    int priority;
    int completed;
} Task;

void add_task(const char *description, int priority) {
    FILE *file = fopen(TASK_FILE, "a");
    if (!file) {
        perror("Could not open file");
        return;
    }
    fprintf(file, "%s,%d,%d\n", description, priority, 0);
    fclose(file);
    printf("Added task: %s\n", description);
}

void list_tasks() {
    FILE *file = fopen(TASK_FILE, "r");
    if (!file) {
        perror("Could not open file");
        return;
    }
    char line[128];
    int index = 1;
    while (fgets(line, sizeof(line), file)) {
        Task task;
        sscanf(line, "%[^,],%d,%d", task.description, &task.priority, &task.completed);
        printf("%d. %s - Priority: %d - Status: %s\n", index++, task.description, task.priority, task.completed ? "Done" : "Pending");
    }
    fclose(file);
}

void complete_task(int task_number) {
    FILE *file = fopen(TASK_FILE, "r");
    if (!file) {
        perror("Could not open file");
        return;
    }
    FILE *temp_file = fopen("temp.txt", "w");
    if (!temp_file) {
        perror("Could not open temporary file");
        fclose(file);
        return;
    }

    char line[128];
    int index = 1;
    while (fgets(line, sizeof(line), file)) {
        Task task;
        sscanf(line, "%[^,],%d,%d", task.description, &task.priority, &task.completed);
        if (index == task_number) {
            task.completed = 1;
            printf("Task %d marked as completed.\n", task_number);
        }
        fprintf(temp_file, "%s,%d,%d\n", task.description, task.priority, task.completed);
        index++;
    }
    fclose(file);
    fclose(temp_file);
    remove(TASK_FILE);
    rename("temp.txt", TASK_FILE);
}

void delete_task(int task_number) {
    FILE *file = fopen(TASK_FILE, "r");
    if (!file) {
        perror("Could not open file");
        return;
    }
    FILE *temp_file = fopen("temp.txt", "w");
    if (!temp_file) {
        perror("Could not open temporary file");
        fclose(file);
        return;
    }

    char line[128];
    int index = 1;
    while (fgets(line, sizeof(line), file)) {
        if (index != task_number) {
            fputs(line, temp_file);
        } else {
            printf("Deleted task %d\n", task_number);
        }
        index++;
    }
    fclose(file);
    fclose(temp_file);
    remove(TASK_FILE);
    rename("temp.txt", TASK_FILE);
}

int main(int argc, char *argv[]) {
    if (argc < 2) {
        printf("Usage:\n");
        printf("  %s add <description> <priority>\n", argv[0]);
        printf("  %s list\n", argv[0]);
        printf("  %s complete <task_number>\n", argv[0]);
        printf("  %s delete <task_number>\n", argv[0]);
        return 1;
    }

    if (strcmp(argv[1], "add") == 0 && argc >= 4) {
        add_task(argv[2], atoi(argv[3]));
    } else if (strcmp(argv[1], "list") == 0) {
        list_tasks();
    } else if (strcmp(argv[1], "complete") == 0 && argc >= 3) {
        complete_task(atoi(argv[2]));
    } else if (strcmp(argv[1], "delete") == 0 && argc >= 3) {
        delete_task(atoi(argv[2]));
    } else {
        printf("Invalid command.\n");
    }

    return 0;
}

File: CLITaskmanager.py
import argparse
import json
import os


TASKS_FILE = 'tasks.json'


def load_tasks():
    if os.path.exists(TASKS_FILE):
        with open(TASKS_FILE, 'r') as f:
            return json.load(f)
    return []


def save_tasks(tasks):
    with open(TASKS_FILE, 'w') as f:
        json.dump(tasks, f, indent=4)


def add_task(description, priority=1):
    tasks = load_tasks()
    tasks.append({'description': description, 'priority': priority, 'completed': False})
    save_tasks(tasks)
    print(f"Added task: {description}")


def list_tasks():
    tasks = load_tasks()
    for i, task in enumerate(tasks, 1):
        status = "Done" if task['completed'] else "Pending"
        print(f"{i}. {task['description']} - Priority: {task['priority']} - Status: {status}")


def complete_task(index):
    tasks = load_tasks()
    if 0 <= index < len(tasks):
        tasks[index]['completed'] = True
        save_tasks(tasks)
        print(f"Task {index + 1} marked as completed.")
    else:
        print("Invalid task number.")


def delete_task(index):
    tasks = load_tasks()
    if 0 <= index < len(tasks):
        removed_task = tasks.pop(index)
        save_tasks(tasks)
        print(f"Deleted task: {removed_task['description']}")
    else:
        print("Invalid task number.")


parser = argparse.ArgumentParser(description="Simple CLI Task Manager")
subparsers = parser.add_subparsers(dest='command')

add_parser = subparsers.add_parser('add', help='Add a new task')
add_parser.add_argument('description', type=str, help='Task description')
add_parser.add_argument('--priority', type=int, default=1, help='Task priority (default: 1)')


list_parser = subparsers.add_parser('list', help='List all tasks')


complete_parser = subparsers.add_parser('complete', help='Mark a task as complete')
complete_parser.add_argument('index', type=int, help='Task number to complete (1-based index)')


delete_parser = subparsers.add_parser('delete', help='Delete a task')
delete_parser.add_argument('index', type=int, help='Task number to delete (1-based index)')


args = parser.parse_args()
if args.command == 'add':
    add_task(args.description, args.priority)
elif args.command == 'list':
    list_tasks()
elif args.command == 'complete':
    complete_task(args.index - 1)
elif args.command == 'delete':
    delete_task(args.index - 1)
else:
    parser.print_help()
